src/compiler/utilities.rs

```diff
@@ -1,19 +1,71 @@
 #![allow(non_upper_case_globals)]
 
+use std::cmp::Ordering;
+
 use crate::{BaseNode, DiagnosticMessage, DiagnosticWithDetachedLocation, SyntaxKind};
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum OperatorPrecedence {
+    Comma,
+    Multiplicative,
+    Primary,
+    Invalid = -1,
+}
+
+impl OperatorPrecedence {
+    pub const Lowest: OperatorPrecedence = OperatorPrecedence::Comma;
+}
+
+impl Ord for OperatorPrecedence {
```

Comment:
In `parseBinaryExpressionRest()` it compares `OperatorPrecedence`'s by their numeric enum value so I implemented `Ord`/`PartialOrd` to be able to write it similarly in Rust (eg `new_precedence > precedence`)

Reply:
I think you could derive rather than implement this if (although you might need to rearrange it so `Invalid` was first)?

Reply:
Right that would be my understanding from reading the [docs](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html#derivable). In this case I'd like to mimic the Typescript ordering with `Invalid` last